store/CSDK/test/main.cpp

```diff
@@ -665,6 +699,208 @@ bool readFromS3(JNIEnv *env, char *path, char *argv[]) {
+        bool result =testWriteData(env, path, my_argc, my_argv);
+        if(result){
+            bool schema_result = readSchema(env, path, false);
+            EXPECT_TRUE(schema_result) << "Not Able to read readSchema from given path";
+        } else {
+            EXPECT_TRUE(result) << "Either Data Loading Or Carbon Reader  is failed";
+        }
+    } catch (runtime_error e) {
+        EXPECT_TRUE(false) << " Exception is not expected ,During Read Schema";
+    }
+}
+
+
+TEST(CSDKTest,readSchemaWithValidation) {
+    try {
+        char *path = "./data_readPathWithValidation";
+        bool result =testWriteData(env, path, my_argc, my_argv);
+        if(result){
+            bool schema_result = readSchema(env, path, true);
+            EXPECT_TRUE(schema_result) << "Not Able to read readSchema Or validate from given path";
+        } else {
+            EXPECT_TRUE(result) << "Either Data Loading Or Carbon Reader  is failed";
+        }
+    } catch (runtime_error e) {
+        EXPECT_TRUE(false) << " Exception is not expected ,During Read Schema";
+    }
+}
+
+TEST(CSDKTest,testReadNextRowWthtVector) {
+    try {
+        int printNum = 32000;
+        char *path = "./data_forVector";
+        bool result =testWriteData(env, path, my_argc, my_argv);
+        if(result){
+            bool readresultWithVector= testReadNextRow(env, path, printNum, my_argv, 0, true);
+            EXPECT_TRUE(readresultWithVector) << "Vector reading is failed";
+        } else {
+            EXPECT_TRUE(result) << "Either Data Loading Or Carbon Reader  is failed";
+        }
+    } catch (runtime_error e) {
+        EXPECT_TRUE(false) << " Exception is not expected ,During Vector read";
+    }
+}
+
+
+TEST(CSDKTest,testReadNextRowWthoutVector) {
+    try {
+        int printNum = 32000;
+        char *path = "./data_forCarbonReader";
+        bool result =testWriteData(env, path, my_argc, my_argv);
+        if(result){
+            bool readresultWithVectorfalse=testReadNextRow(env, path, printNum, my_argv, 0, false);
+            EXPECT_TRUE(readresultWithVectorfalse) << "Carbon Reader (Vector=false) is failed";
+        } else {
+            EXPECT_TRUE(result) << "Either Data Loading Or Carbon Reader  is failed";
+        }
+    } catch (runtime_error e) {
+        EXPECT_TRUE(false) << " Exception is not expected ,During Carbon Read";
+    }
+}
+
+
+TEST(CSDKTest,testReadNextBatchRowWithVector) {
+    try {
+        int printNum = 32000;
+        int batch = 32000;
+        char *path = "./data_forCarbonReader";
+        bool result =testWriteData(env, path, my_argc, my_argv);
+        if(result){
+            bool readresultWithVector=testReadNextBatchRow(env, path, batch, printNum, my_argv, 0, true);
+            EXPECT_TRUE(readresultWithVector) << "Vector Reader With Batch  is failed";
+        } else {
+            EXPECT_TRUE(result) << "Either Data Loading Or Carbon Reader  is failed";
+        }
+    } catch (runtime_error e) {
```

Comment:
Should we log the runtime_error to observe the failures in CI ?

Reply:
Handled comments:-
1. added method convertExceptionToString for each throw which will print the exception 
2. env->ExceptionDescribe();   method already prints the Exception trace. 
3. Defined message after <<  in EXPECT_TRUE  is assert message.